### PR TITLE
fix(docs): Fix serialisation errors for property groups

### PR DESCRIPTION
**Pull request by neilkakkar**
## Problem

Fixes sentry errors around property validation

## Changes

There was no easy way to represent recursive Serializers that could either be a property or a property group... And it also needs to support old style properties. Gah!

Open to other ideas, but in the end I decided to just create a custom field that 'borrows' implementation from the property mixin.

This does mean that the property documentation can get out of date, since there aren't two places to update code anymore, but seems like an okay tradeoff for the uber complicated input fields.

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?

Run locally, see that no exception is raised for funnel and trends while serializing request

### CODE REVIEW

File: posthog/api/insight_serializers.py [hunk 1]
```diff
@@ -51,7 +51,26 @@ def __init__(self, request=None, *args, **kwargs):
...
+        where a property looks like:
+        {
+            "key": "email",
+            "value": "x@y.com",
+            "operator": "exact",
+            "type": "event"
+        }.
+
+        Can be nested arbitrarily.
+        """,
```
Comment (timgl) [inline posthog/api/insight_serializers.py:72]: You can use markdown here! Also you don't want to indent this stuff
```suggestion
        help_text="""Properties to filter on. A JSON object that looks like:
```json
{
    "type": "AND",
    "values": [ Properties ]
}
```

where a property looks like:
```json
{
    "key": "email",
    "value": "x@y.com",
    "operator": "exact",
    "type": "event"
}
```

Can be nested arbitrarily.""",
```

**Review by EDsCODE — APPROVED** [review]
let's go with this for now

**Comment by timgl** [comment]
@neilkakkar @EDsCODE See my latest commit. The problem with the earlier solution was
1. it get's rid of all of this useful context which is auto updated if we add more options
 
image
2. No validation.

My proposed solution should fix both. In the future, you can check how any changes affect the api docs by looking at http://127.0.0.1:8000/api/schema/redoc/#operation/Trends

File: posthog/api/insight.py [hunk 2]
```diff
@@ -296,11 +296,8 @@ def layouts(self, request, **kwargs):
     )
     @action(methods=["GET", "POST"], detail=False)
     def trend(self, request: request.Request, *args: Any, **kwargs: Any):
-        try:
-            serializer = TrendSerializer(request=request)
-            serializer.is_valid(raise_exception=True)
-        except Exception as e:
-            capture_exception(e)
+        serializer = TrendSerializer(request=request)
```
Comment (neilkakkar) [inline posthog/api/insight.py:299]: error?

**Comment by neilkakkar** [comment]
This makes sense, It doesn't need to be an accurate representation of the data model, ha! As long as there's some comment explaining other possibilities.

Nice, I like it :)

That link to redoc is massively helpful, thanks!

**Comment by timgl** [comment]
Let's wait merging this until after code freeze

**Pull request merged**